server/src/main/java/com/thoughtworks/go/server/service/PluginService.java

```diff
@@ -194,4 +199,12 @@ private boolean isRequestFresh(String md5, PluginSettings pluginSettings, Locali
 
     }
 
+    private PluginInfo pluginInfoFor(String pluginId) {
+        final PluginInfo pluginInfo = pluginInfoFinder.pluginInfoFor(pluginId);
+        if (pluginInfo == null) {
+            throw new RuntimeException(String.format("The plugin with id %s is not loaded.", pluginId));
```

Comment:
We will end up changing the behaviour by doing this. In absence of the plugin, can users fetch the plugin settings? I know we cannot update in absence of the plugin.

Reply:
No in absence of plugin settings we cannot get plugin settings since we don't have information about plugin settings metadata.  Without plugin settings metadata we don't know whether the property is secure or not.